### ISSUE TITLE
v1.9.0 crashing on `terraform plan`

### ISSUE THREAD

**Issue by DavidS-ovm** [vercel/terraform-provider-vercel#175]
Hi,

after months of smooth usage, when trying to apply the latest update, I've started to see crashes during terraform planning in our CI. Reverting to 1.8.0 has cleared that up again.

```
Error: Request cancelled
The plugin6.(*GRPCProvider).ReadDataSource request was cancelled.
Error: Request cancelled
The plugin6.(*GRPCProvider).ValidateDataResourceConfig request was cancelled.
Releasing state lock. This may take a few moments...
data.vercel_project.frontend: Still reading... [50s elapsed]
Stack trace from the terraform-provider-vercel_v1.9.0 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1 pc=0xa6692a]
goroutine 39 [running]:
github.com/vercel/terraform-provider-vercel/vercel.convertResponseToProjectDataSource(***_, _***, ***0x0, 0xc00035e660, 0x0, 0xc00035e670, ***0xc0003da3c0, 0x20***, ***0xc0001299a0, 0x1d***, ...***, ...)
	github.com/vercel/terraform-provider-vercel/vercel/data_source_project.go:318 +0x10a
github.com/vercel/terraform-provider-vercel/vercel.(*projectDataSource).Read(0xc00012c048, ***0xd88168, 0xc0001d5b60***, ***0xd8c598, 0xc000332750***, ***0xb55240, 0xc0002f7c50***, ***0xd8ecc0, 0xc000114080***, ***0x0, ...***, ...***, ...***, ...)
	github.com/vercel/terraform-provider-vercel/vercel/data_source_project.go:397 +0x591
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadDataSource(0xc0001f3040, ***0xd88168, 0xc0001d5b60***, 0xc0001d5bf0, 0xc0002656c8)
	github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/fwserver/server_readdatasource.go:79 +0x433
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadDataSource(0xc0001f3040, ***0xd88168?, 0xc0001d5a40?***, 0xc00012ea40)
	github.com/hashicorp/terraform-plugin-framework@v1.7.0/internal/proto6server/server_readdatasource.go:55 +0x41c
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadDataSource(0xc0001cc640, ***0xd88168?, 0xc0001b0990?***, 0xc00012a5f0)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/tf6server/server.go:686 +0x417
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadDataSource_Handler(***0xc280c0?, 0xc0001cc640***, ***0xd88168, 0xc0001b0990***, 0xc000118780, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:572 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000197000, ***0xd88168, 0xc0001b0330***, ***0xd8d360, 0xc0001a9e00***, 0xc00015b440, 0xc0002f7500, 0x127ae20, 0x0)
	google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000197000, ***0xd8d360, 0xc0001a9e00***, 0xc00015b440)
	google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 16
	google.golang.org/grpc@v1.63.2/server.go:1030 +0x135
Error: The terraform-provider-vercel_v1.9.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Error: Process completed with exit code 1.
```

----
Versions:
```
Terraform v1.5.7
on linux_amd64
+ provider registry.terraform.io/alekc/kubectl v2.0.4
+ provider registry.terraform.io/auth0/auth0 v1.2.0
+ provider registry.terraform.io/cyrilgdn/postgresql v1.22.0
+ provider registry.terraform.io/gavinbunney/kubectl v1.14.0
+ provider registry.terraform.io/go-gandi/gandi v2.3.0
+ provider registry.terraform.io/hashicorp/aws v5.47.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.3
+ provider registry.terraform.io/hashicorp/googleworkspace v0.7.0
+ provider registry.terraform.io/hashicorp/helm v2.13.1
+ provider registry.terraform.io/hashicorp/http v3.4.2
+ provider registry.terraform.io/hashicorp/kubernetes v2.29.0
+ provider registry.terraform.io/hashicorp/local v2.5.1
+ provider registry.terraform.io/hashicorp/null v3.2.2
+ provider registry.terraform.io/hashicorp/random v3.6.1
+ provider registry.terraform.io/hashicorp/time v0.11.1
+ provider registry.terraform.io/hashicorp/tls v4.0.5
+ provider registry.terraform.io/integrations/github v6.2.1
+ provider registry.terraform.io/mietz-gmbh/process v0.1.3
+ provider registry.terraform.io/vercel/vercel v1.9.0
```

**Comment by dglsparsons** [comment]
He @DavidS-ovm, thanks for raising this. It looks like a provider bug around the `vercel_project` data source. - I'll investigate and hopefully have a fix out soon 🙏

**Comment by DavidS-ovm** [comment]
Thanks for the quick reply.

> It looks like a provider bug around the `vercel_project` data source.

Yup, we're using the `vercel_project` data source.

**Comment by dglsparsons** [comment]
Hi @DavidS-ovm, this should be fixed in [v1.9.1](https://github.com/vercel/terraform-provider-vercel/releases/tag/v1.9.1) of the provider. Please upgrade and feel free to re-open this if the issue continues 🙏

**Comment by DavidS-ovm** [comment]
A slight improvement:
```
Error: Value Conversion Error
  with data.vercel_project.frontend,
  on frontend.tf line 18, in data "vercel_project" "frontend":
  18: data "vercel_project" "frontend" ***
An unexpected error was encountered trying to convert from struct into an
object. This is always an error in the provider. Please report the following
to the provider developer:
Mismatch between struct and object type: Struct defines fields not found in
object: deploy_hooks.
Struct: vercel.GitRepository
Object type: types.ObjectType["production_branch":basetypes.StringType,
"repo":basetypes.StringType, "type":basetypes.StringType]
Error: Value Conversion Error
  with data.vercel_project.docs-overmind-tech,
  on frontend.tf line 22, in data "vercel_project" "docs-overmind-tech":
  22: data "vercel_project" "docs-overmind-tech" ***
An unexpected error was encountered trying to convert from struct into an
object. This is always an error in the provider. Please report the following
to the provider developer:
Mismatch between struct and object type: Struct defines fields not found in
object: deploy_hooks.
Struct: vercel.GitRepository
Object type: types.ObjectType["production_branch":basetypes.StringType,
"repo":basetypes.StringType, "type":basetypes.StringType]
Releasing state lock. This may take a few moments...
Error: Process completed with exit code 1.
```

**Comment by DavidS-ovm** [comment]
PS: I don't have enough permissions on this repo to re-open issues

**Comment by dglsparsons** [comment]
Oh no. I'm not sure why i can't reproduce these issues myself. I think you perhaps have an old project that doesn't match what the API is expected to return.

**Comment by dglsparsons** [comment]
Ah, i was able to reproduce this one. Fix coming soon.

**Comment by dglsparsons** [comment]
Okay @DavidS-ovm, that should be fixed in [v1.9.2](https://github.com/vercel/terraform-provider-vercel/releases/tag/v1.9.2). 🙏 
Again, just let me know if there are any issues.
